Libraries/Components/View/ViewPropTypes.js

```diff
@@ -61,18 +68,42 @@ type DirectEventProps = $ReadOnly<{|
    *
    * See http://facebook.github.io/react-native/docs/view.html#onmagictap
    */
-  onMagicTap?: ?Function,
+  onMagicTap?: ?(event: SyntheticEvent<null>) => mixed,
+|}>;
+
+type Touch = $ReadOnly<{|
+  locationX: number,
+  locationY: number,
+  pageX: number,
+  pageY: number,
+  screenX: number,
+  screenY: number,
+  identifier: number,
+  target: number,
+  timestamp: number,
+  force: number,
 |}>;
 
+type TouchEvent = SyntheticEvent<
```

Comment:
It looks like when I typed PressEvent I didn't know most of that would only get assigned if Fabric was being used. I will have to look more closely and see if most of that is relevant to `PressEvent` and if I should slim it down, or remove `TouchEvent` for being redundant (and rename `PressEvent` to `TouchEvent`, perhaps).

It looks like these might be the same event props as the ones on ScrollView, but it looks like all Views might be getting these events. I will see if I can get some information about these props in #fabric

Reply:
I believe that these are things that all views get and should probably only appear in ViewProps. ScrollView and RCTText all inherit from View's props so they should all have the same types for the responders.

In Flow ScrollView is actually spreading ViewProps so it doesn't even need to be defined again there. The text props are a bit more complicated since they aren't actually spreading ViewProps right now so I'd just make sure they are the same types in both ViewProps and TextProps.

Reply:
I think that `TouchEvent` might be redundant with `PressEvent` (thanks for pointing that out!) I will remove it for now, and could revisit it when Fabric is in OSS